File: Samples/Legacy/SSSExampleVBnet/SSSExampleVBnet/WriteText/Form1.vb
Imports ScreenScraper

Public Class Form1
    Private m_tCapture As TextCaptureX
    Private m_tSelection As TSelection
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        radioAPI.Checked = True
        Try
            m_tCapture = ComFactory.Instance.NewTextCaptureX
            m_tSelection = ComFactory.Instance.NewTSelection
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim tSelInfo As TSelectionInfo
        Try
            tSelInfo = m_tSelection.Start(TS_SELECTION.tsSelectionWindow, TS_SELECTION_OPTIONS.tsSelFlagDefaultText)

        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try
        Dim strID As String
        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem
            uiElement.hwnd = tSelInfo.WindowHandle
            strID = uiElement.GetID(True)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try
        txtHandle.Text = uiElement.hwnd.ToString("X")
        txtID.Text = strID

    End Sub

    Private Sub btnRectangle_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRectangle.Click
        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem
            uiElement.UseClientCoordinates = True
            uiElement.InitializeFromID(txtID.Text, True)

            Dim left, top, right, bottom As Integer
            uiElement.GetRectangle(left, top, right, bottom)

            Dim flags As Integer
            flags = UIE_ClickFlags.UIE_CF_MOVE_CURSOR Or UIE_ClickFlags.UIE_CF_SINGLE

            Dim method As Integer
            method = 0
            If (radioAPI.Checked = True) Then
                method = UIE_WriteTextMethod.UIE_WTM_NATIVE
            Else
                method = UIE_WriteTextMethod.UIE_WTM_SENDKEYS
            End If

            uiElement.WriteText(method, txtWriteText.Text)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub
End Class

File: Samples/GUI Automation API/Legacy/SSSExampleCS/SSSExampleCS/WriteText/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

using ScreenScraper;

namespace WriteText
{
    public partial class Form1 : Form
    {
        TextCaptureX m_tCapture;
        TSelection m_tSelection;

        public Form1()
        {
            InitializeComponent();

            radioAPI.Checked = true;

            try
            {
                m_tCapture = ComFactory.Instance.NewTextCaptureX();
                m_tSelection = ComFactory.Instance.NewTSelection();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            TSelectionInfo tSelInfo = null;

            this.WindowState = FormWindowState.Minimized;
            try
            {
                tSelInfo = (TSelectionInfo)m_tSelection.Start((int)TS_SELECTION.tsSelectionUIElement, (int)TS_SELECTION_OPTIONS.tsSelFlagDefaultText);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }
            this.WindowState = FormWindowState.Normal;
            string strID = null;
            UIElem uiElem = null;
            try
            {
                uiElem = ComFactory.Instance.NewUIElem();
                uiElem.InitializeFromID(tSelInfo.UIElementID, false);
                strID = tSelInfo.UIElementID;
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }

            txtHandle.Text = uiElem.hwnd.ToString("X");
            txtID.Text = strID;
        }

        private void btnRectangle_Click(object sender, EventArgs e)
        {
            UIElem uiElem = null;
            try
            {
                uiElem = ComFactory.Instance.NewUIElem();
                uiElem.UseClientCoordinates = false;
                uiElem.InitializeFromID(txtID.Text, false);

                // click on the element
                int flags = (int)UIE_ClickFlags.UIE_CF_MOVE_CURSOR | (int)UIE_ClickFlags.UIE_CF_SINGLE | (int)UIE_ClickFlags.UIE_CF_LEFT;
                uiElem.Activate();
                uiElem.Click(2, 2, flags);

                // write text
                int method = 0;
                if (radioAPI.Checked == true)
                {
                    method = (int)UIE_WriteTextMethod.UIE_WTM_NATIVE;
                }
                else if (radioSendKeys.Checked == true)
                {
                    method = (int)UIE_WriteTextMethod.UIE_WTM_SENDKEYS;
                }
                string strErase = "[k(end)d(shift)k(home)u(shift)k(del)]";
                uiElem.WriteText(method, strErase + txtWriteText.Text);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }
    }
}
